connect-inject/endpoints_controller.go

```diff
@@ -473,10 +472,11 @@ func (r EndpointsController) filterAgentPods(meta metav1.Object, object runtime.
 // are on the same node as the new Consul Agent pod. It receives a Pod Object which is a
 // Consul Agent that has been filtered by filterAgentPods and only enqueues endpoints
 // for client agent pods where the Ready condition is true.
-func (r EndpointsController) requestsForRunningAgentPods(object handler.MapObject) []ctrl.Request {
+func (r EndpointsController) requestsForRunningAgentPods(object client.Object) []ctrl.Request {
 	var consulClientPod corev1.Pod
-	r.Log.Info("received update for consulClientPod", "podName", object.Meta.GetName())
-	err := r.Client.Get(r.Ctx, types.NamespacedName{Name: object.Meta.GetName(), Namespace: object.Meta.GetNamespace()}, &consulClientPod)
+	ctx := context.Background()
+	r.Log.Info("received update for consulClientPod", "podName", object.GetName())
+	err := r.Client.Get(ctx, types.NamespacedName{Name: object.GetName(), Namespace: object.GetNamespace()}, &consulClientPod)
```

Comment:
Previously, we were using the same context from the controller, but now we can't I guess because we don't have it yet? Could we pass it from the manager so that it's still one context for the controller?

Reply:
yeah.. i was trying to think of a way of attempting to re-use it from the manager, but also wanted to avoid having the Reconciler pass around a context and also have one defined in the struct. let me revisit this real quick

Reply:
added back context but I only used it in that one function. I would like Reconcile to use the context that is passed to it from the machinery tho

Reply:
I do like not having to create a new context here though 😄